src/util/device_path.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 CoreOS, Inc.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use base::Status;
+use protocol::{DevicePathProtocol, DevicePathUtilitiesProtocol};
+use void::CVoid;
+
+pub fn create_file_device_node(filename: &str) -> Result<&DevicePathProtocol, Status> {
+    let node_size_bytes = 4 + (filename.len() + 1) * 2;
```

Comment:
Convert the filename with `str_to_utf16_ptr` and then use `utf16_strlen`.  The length is used in unsafe code and may change in the future (if surrogate pairs are ever supported) so you'll want to avoid making assumptions here.